rest.go

```diff
@@ -90,39 +51,39 @@ func BuildResponse(res *http.Response) (*Response, error) {
 }
 
 // API is the main interface to the API.
-func API(request Request) (*Response, error) {
+func API(request *http.Request) (*Response, error) {
 	return DefaultClient.API(request)
 }
 
 // The following functions enable the ability to define a
 // custom HTTP Client
 
 // MakeRequest makes the API call.
-func (c *Client) MakeRequest(req *http.Request) (*http.Response, error) {
+func (c *Client) makeRequest(req *http.Request) (*http.Response, error) {
 	return c.HTTPClient.Do(req)
 }
 
 // API is the main interface to the API.
-func (c *Client) API(request Request) (*Response, error) {
-	// Add any query parameters to the URL.
-	if len(request.QueryParams) != 0 {
-		request.BaseURL = AddQueryParameters(request.BaseURL, request.QueryParams)
+func (c *Client) API(request *http.Request) (*Response, error) {
+	if request == nil {
+		return nil, errors.New("invalid request: request cannot be nil")
 	}
+	_, supportedMethod := supportedMethods[request.Method]
 
-	// Build the HTTP request object.
-	req, err := BuildRequestObject(request)
-	if err != nil {
-		return nil, err
+	if !supportedMethod {
+		return nil, errors.New("unsupported request method:" + request.Method)
+	}
```

Comment:
Sorry for the delayed response. I only added this for reverse compatibility with what was done before. Because of the existing Request abstraction only supported methods were even permissible changing the API to take vanilla http.Request instances removed that possibility. Great comments otherwise. I'll make the suggested changes and update the PR.